File: Problems/51866.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int a, b;
    cin >> a >> b;

    vector<int> v;
    int t;

    for (int i = 0; i < a; i++) {
        cin >> t;
        v.push_back(t);
    }

    int lv = v.size();
    int max = 0;

    for (int i = 0; i < lv; i++) {
        if (v[i] > max) {
            max = v[i];
        }
    }
    int min = 5001;
    for (int i = 0; i < lv; i++) {
        if (min > v[i]) {
            min = v[i];
        }
    }

    if (b == 1 or (b == 3 and a <= 3)) {
        cout << max;
    }else if (b == 3 and a > 3) {
        cout << min;
    }

    
    else if (b == 2) {
    int smallest = 5001;

    for (int i = 1; i < lv; i++) {
        if (v[i] < v[i - 1] && v[i] < smallest) {
            smallest = v[i];
        }
    }

    cout << smallest;
}
    
}

File: Problems/51866.py
n = list(map(int, input().strip().split()))
v = list(map(int, input().strip().split()))
def cal():
    if n[1] == 1:
        print(max(v))
    elif n[1] >= 3:
        print(min(v))
    else:
        m = min(v)
        if v[0] == m or v[-1] == 0:
            print(m)
        else:
            print(min((v[0], v[-1])))
cal()